### PR TITLE
feat: add backup callback to redirect auth RELEASE

### PR DESCRIPTION
**Pull request by itaihanski**
## Related Issues

https://github.com/descope/etc/issues/5062

## Description

Add new backup callback redirect auth parameter

## Must

- [X] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 1]
```diff
@@ -179,6 +180,7 @@ class DescopeWc extends BaseDescopeWc {
         ? {
             callbackUrl: redirectAuthCallbackUrl,
             codeChallenge: redirectAuthCodeChallenge,
+            backupCallbackUri: redirectAuthBackupCallbackUri,
```
Comment (asafshen) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:183]: is there is a chance that `redirectAuthBackupCallbackUri` exists and `redirectAuthCallbackUrl` does not exists?

image

File: packages/web-component/src/lib/helpers/helpers.ts [hunk 2]
```diff
@@ -267,6 +275,7 @@ export const handleUrlParams = () => {
     exchangeError,
     redirectAuthCodeChallenge,
     redirectAuthCallbackUrl,
+    redirectAuthBackupCallbackUri,
```
Comment (asafshen) [inline packages/web-component/src/lib/helpers/helpers.ts:278]: not critical, naming wise, I understand that technically this is a URI, but having the `redirectAuthCallbackUrl` (which is also URI_ named with `Url`, I would rename the backup callback also "url" and not "uri" for consistency 

wdyt?

**Review by asafshen — COMMENTED** [review]
looks good, left 2 questions

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 1 above)
Comment (itaihanski) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts:183]: Good question - It's not allowed

File: packages/web-component/src/lib/helpers/helpers.ts (hunk 2 above)
Comment (itaihanski) [reply packages/web-component/src/lib/helpers/helpers.ts:278]: This is intentional - the URL is actually an http/https based url (app link / universal link). The backup URI is however a custom scheme URI, used only on Android + Opera at the moment. This is to distinguish between them

Comment (asafshen) [reply packages/web-component/src/lib/helpers/helpers.ts:278]: cool 👍 
as far as I know the fact that there is a different scheme (aka myapp:// instead https://) doesn't affect this terminology between url to uri but if it make sense to you its enough

**Review by asafshen — APPROVED** [review]
great 👍

**Pull request merged**